Remotion/Core/Core/Reflection/TypeExtensions.cs

```diff
@@ -112,53 +89,38 @@ private static bool CanAscribeInternal (Type type, Type ascribeeType)
       Type ascribeeGenericTypeDefinition = ascribeeType.GetGenericTypeDefinition();
       for (Type? currentType = type; currentType != null; currentType = currentType.BaseType)
       {
-        if (CanDirectlyAscribeToGenericTypeInternalFromCache (currentType, ascribeeType, ascribeeGenericTypeDefinition))
+        if (CanDirectlyAscribeToGenericTypeInternal (currentType, ascribeeType, ascribeeGenericTypeDefinition))
           return true;
       }
       return false;
     }
 
-    private static IReadOnlyList<Type> GetAscribedGenericArgumentsWithoutCache (Type type, Type ascribeeType)
-    {
-      if (!ascribeeType.IsGenericType)
-      {
-        if (ascribeeType.IsAssignableFrom (type))
-          return Type.EmptyTypes;
-        else
-          throw ArgumentUtility.CreateArgumentTypeException ("type", type, ascribeeType);
-      }
-      else if (ascribeeType.IsInterface)
-        return GetAscribedGenericInterfaceArgumentsInternal (type, ascribeeType);
-      else
-        return GetAscribedGenericClassArgumentsInternal (type, ascribeeType);
```

Comment:
please use this block (and add missing braces) instead of dropping the ELSE notiations. 

Reasoning: IF/ELSE is better readable when not doing alternative code branches instead of having a single main branch and several early-exit conditions.

Reply:
hmm ok, how about a `switch statement` or a `switch expression` to get rid of the `if`s and `else`s?
e.g.
```Csharp

return ascribeeType switch
{
    { IsGenericType: false } when ascribeeType.IsAssignableFrom (type) => Type.EmptyTypes,
    { IsGenericType: false } => throw ArgumentUtility.CreateArgumentTypeException ("type", type, ascribeeType),
    _ => ascribeeType.IsInterface
            ? GetAscribedGenericInterfaceArgumentsInternal (type, ascribeeType)
            : GetAscribedGenericClassArgumentsInternal (type, ascribeeType)
};
```

Reply:
no, just keep the original, please (refactoring of this is out of scope for the current task).  

Also: switch (or, rather pattern matching) should be used for actual pattern cases, not just as an alternative to boolean conditions in an IF. The block structure is actually helpful in the above statement.